### PR TITLE
[Backport 2.x] Add release notes for v2.6.0 (#6443)

### PR DESCRIPTION
**Pull request by kartg**
Signed-off-by: Kartik Ganesh <gkart@amazon.com>
(cherry picked from commit 01466a6a04e280267f013aeb14dff1f20dc7a77a)

### Description

* Add release notes for v2.6.0

This change also includes cleanup of the CHANGELOG file to only capture unreleased changes.

Signed-off-by: Kartik Ganesh <gkart@amazon.com>

* Minor language updates in v2.6.0 release notes and Changelog

Signed-off-by: Kartik Ganesh <gkart@amazon.com>

### Issues Resolved
relates #6443 and #6360 

### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [ ] Commits are signed per the DCO using --signoff
- [ ] Commit changes are listed out in CHANGELOG.md file (See: [Changelog](../blob/main/CONTRIBUTING.md#changelog))

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: release-notes/opensearch.release-notes-2.6.0.md [hunk 1]
```diff
@@ -0,0 +1,57 @@
+## 2023-02-22 Version 2.6.0 Release Notes
+
+## [2.6]
+### Added
+- Add index create block when all nodes have breached high disk watermark ([#5852](https://github.com/opensearch-project/OpenSearch/pull/5852))
+- Add cluster manager throttling stats in nodes/stats API ([#5790](https://github.com/opensearch-project/OpenSearch/pull/5790))
+- Add support for feature flags in opensearch.yml ([#4959](https://github.com/opensearch-project/OpenSearch/pull/4959))
+- Add query for initialized extensions ([#5658](https://github.com/opensearch-project/OpenSearch/pull/5658))
```
Comment (owaiskazi19) [inline release-notes/opensearch.release-notes-2.6.0.md:8]: We should remove this as it's part of extensibility.

Comment (andrross) [reply release-notes/opensearch.release-notes-2.6.0.md:8]: For any updates of the release notes we should start with a PR on main and then backport them to the relevant branches.

**Review by andrross — APPROVED** [review]

File: release-notes/opensearch.release-notes-2.6.0.md (hunk 1 above)
Comment (andrross) [reply release-notes/opensearch.release-notes-2.6.0.md:8]: @owaiskazi19 Feel free to make a PR on main for this and any other edits

Comment (kartg) [reply release-notes/opensearch.release-notes-2.6.0.md:8]: As per @andrross 's comment, i'll create a PR to `main` to remove these. Are these the only 4 lines to remove?

Comment (kartg) [reply release-notes/opensearch.release-notes-2.6.0.md:8]: @owaiskazi19 please go ahead and create a PR for these changes, or let me know if you'd like me to

Comment (owaiskazi19) [reply release-notes/opensearch.release-notes-2.6.0.md:8]: I was referring to just this one line.

Comment (owaiskazi19) [reply release-notes/opensearch.release-notes-2.6.0.md:8]: PR for the same: https://github.com/opensearch-project/OpenSearch/pull/6451

**Pull request merged**